test/odbc/test_pg_datatype.cpp

```diff
@@ -0,0 +1,160 @@
+#include <gtest/gtest.h>
+#include <sqlext.h>
+#include "odbc_handler.h"
+#include "query_generator.h"
+
+using std::pair;
+
+class PSQL_DataTypes : public testing::Test{
+
+};
+
+TEST_F(PSQL_DataTypes, Datatypes_bigint) {
+
+  const string TABLE_NAME = "master_dbo.bigint_table";
+  const string COL1_NAME = "pk"; // primary key
+  const string COL2_NAME = "data";
+  const string DATATYPE_NAME = "sys.bigint";
+
+  const int BYTES_EXPECTED = 8;
+  const int NUM_ROWS = 4;
+  const int LENGTH_EXPECTED = 20;
+  const int PRECISION_EXPECTED = 0;
+  const int SCALE_EXPECTED = 0;
+  const int BUFFER_SIZE = 256;
+  const string NAME_EXPECTED = "int8";
+  
+  RETCODE rcode;
+  OdbcHandler odbcHandler;
+  SQLLEN pk_len;
+  SQLLEN data_len;
+  long long int pk;
+  long long int data;
+  int length;
+  int precision;
+  int scale;
+  char name[BUFFER_SIZE];
+  int strlength;
+
+  vector<pair<string, string>> BIGINT_TABLE_COLUMNS = {
+    {COL1_NAME, DATATYPE_NAME + " PRIMARY KEY"},
+    {COL2_NAME, DATATYPE_NAME}
+  };
+
+  vector<tuple<int, int, SQLPOINTER, int, SQLLEN* >> bind_columns = {
+
+    {1, SQL_C_SBIGINT, &pk, 0, &pk_len},
+    {2, SQL_C_SBIGINT, &data, 0,  &data_len}
+  };
+
+  vector <string> inserted_values = {
+    "-9223372036854775808",
+    "9223372036854775807",
+    "3",
+    "NULL"
+  };
+
+  string insert_string = ""; 
+  for (int i = 0; i< NUM_ROWS; ++i) {
+    insert_string = insert_string + "(" + std::to_string(i) + "," + inserted_values[i] + ")";
+    if (i+1 != NUM_ROWS)
+      insert_string = insert_string + ", ";
+  }
+
+  // Step 1: Create a table with columns defined with the specific datatype being tested. 
+  odbcHandler.ConnectAndExecQuery(CreateTableStatement(TABLE_NAME, BIGINT_TABLE_COLUMNS));
+
+  // Step 2: Insert values into the table using the correct ODBC data type mapping.
+  odbcHandler.ExecQuery(InsertStatement(TABLE_NAME, insert_string));
+  odbcHandler.CloseStmt();
+
+  // Step 3: Select all from the tables and assert that the following attributes of the type is correct:
+  odbcHandler.ExecQuery(SelectStatement(TABLE_NAME, {"*"}, vector<string> {"pk"}));
+
+  // Make sure attributes are correct
```

Comment:
Maybe we can describe what we're exactly checking better. There are a lot of asserts going on in this 1 test, it took a while to wrap my head around what we were asserting exactly here. 

Similarly for line 114, `// Make sure values are correct` this could be more descriptive. 

Also, I think you're doing this anyway, it'd help to group similar checks/asserts together for organization's sake.

Reply:
Sounds good, I can make it more descriptive.

Reply:
Actually, I've been thinking a lot about organization of these tests. So far we have 1 test that does multiple things & multiple asserts. This can get long & messy, even with comments. 

From what I see you're doing 3 different types of asserts in this 1 test: asserting column metadata, asserting table rows, and asserting table rows after an `UPDATE` statement. Could we separate these into 3 tests? 

What I'm thinking is we'd have a `SetUp()` in the test fixture that creates the table(s) & populates them with data, then the tests themselves, and a `TearDown()` that drops this table at the end. Do you think this is feasible?